mozci/push.py

```diff
@@ -343,6 +367,68 @@ def add(result):
 
         return [Task.create(**task) for task in normalized_tasks]
 
+    def gather_missing_results(self, tasks: list) -> list:
+        """ Fetch and cache errors and results from Taskcluster. Cache together AD and TC results."""
+        test_tasks = {}  # Structure to be cached
+        tasks_missing_results = []  # List of tasks without data from AD
+
+        # Let's gather all collected data from AD
+        for t in tasks:
+            if isinstance(t, TestTask):
+                if not (t._results is None):
+                    # Data for this task has been gathered
+                    test_tasks[t.id] = {
+                        "errors": t._errors,
+                        "results": t._results,
+                    }
+                else:
+                    tasks_missing_results.append(t)
+
+        logger.info("Obtained {} errors/results from AD".format(len(test_tasks.keys())))
+
+        # If there are any test tasks that do not have errors/results from AD let's fetch them
+        # If task.results is not initialized it will be fetched via Taskcluster
+        # Calling task.results modifies the properties of the task, thus, the returning list
+        # of this function comes back modified
+        future_to_task = {
+            Push.THREAD_POOL_EXECUTOR.submit(lambda task: task.results, task): task
+            for task in tasks_missing_results
+        }
+
+        for future in concurrent.futures.as_completed(future_to_task):
+            task = future_to_task[future]
+            logger.debug("Fetching errorsummary for {}".format(task.id))
+            # We assert that the information was not gathered via AD already
+            assert test_tasks.get(task.id) is None
+            # This test task now has the values for errors & results
+            test_tasks[task.id] = {
+                "errors": task.errors,
+                "results": task.results,
+            }
+
+        logger.info(
+            "Obtained {} errors/results from Taskcluster".format(
+                len(tasks_missing_results)
+            )
+        )
```

Comment:
We'd be logging to much with this, can you either remove it or make it debug level?